src/widgets/BankDetailsSummaryRenderWidget.php

```diff
@@ -0,0 +1,69 @@
+<?php
+declare(strict_types=1);
+
+namespace hipanel\modules\client\widgets;
+
+use hipanel\modules\client\models\BankDetails;
+use yii\base\Widget;
+
+class BankDetailsSummaryRenderWidget extends Widget
+{
+    /** @var BankDetails[] */
+    public array $models;
+
+    public function run()
+    {
+        $result = [];
+        foreach ($this->models as $model) {
+            $result[] = implode("\n",
+                array_filter([
+                    $this->renderBankAccount($model),
+                    $this->renderBankName($model->bank_name),
+                    $this->renderBankAddress($model->bank_address),
+                    $this->renderBankSwift($model->bank_swift),
+                    $this->renderCorrespondentBank($model->bank_correspondent),
+                    $this->renderCorrespondentBankSwift($model->bank_correspondent_swift),
+                ]));
+        }
+
+        return implode("\n\n", $result);
+    }
+
+    private function renderBankAccount(BankDetails $model): ?string
+    {
+        if (empty($model->bank_account)) {
+            return null;
+        }
+        $currency = '';
+//        $currency = mb_strtoupper($model->currency);  // todo: adjust currency display
```

Comment:
Currency seems to be always an empty string

Reply:
so far it's meant to be